docs/sources/mimir/release-notes/v2.15.md

```diff
@@ -0,0 +1,128 @@
+
+The ingester can now build 24h blocks for out-of-order data which is >24h old, using the setting `-blocks-storage.tsdb.bigger-out-of-order-blocks-for-old-samples`.
+
+The ruler now supports caching the contents of rule groups via the setting `-ruler-storage.cache.rule-group-enabled`.
+
+The distributor now supports promotion of OTel resource attributes to labels via the setting `-distributor.promote-otel-resource-attributes`.
+
+## Bug fixes
+
+- Alerts: Fix autoscaling metrics joins in `MimirAutoscalerNotActive` when series churn.
+- Alerts: Exclude failed cache "add" operations from alerting since failures are expected in normal operation.
+- Alerts: Exclude read-only replicas from `IngesterInstanceHasNoTenants` alert.
+- Alerts: Use resident set memory for the `EtcdAllocatingTooMuchMemory` alert so that ephemeral file cache memory doesn't cause the alert to misfire.
+- Dashboards: Fix autoscaling metrics joins when series churn.
+- Distributor: Fix pooling buffer reuse logic when `-distributor.max-request-pool-buffer-size` is set.
+- Ingester: Fix issue where active series requests error when encountering a stale posting.
+- Ingester: Fix race condition in per-tenant TSDB creation.
+- Ingester: Fix race condition in exemplar adding.
+- Ingester: Fix race condition in native histogram appending.
+- Ingester: Fix bug in concurrent fetching where a failure to list topics on startup would cause to use an invalid topic ID (0x00000000000000000000000000000000).
+- Ingester: Fix data loss bug in the experimental ingest storage when a Kafka Fetch is split into multiple requests and some of them return an error.
+- Ingester: Fix bug where chunks could have one unnecessary zero byte at the end.
+- OTLP: Support integer exemplar value type.
+- OTLP receiver: Preserve colons and combine multiple consecutive underscores into one when generating metric names in suffix adding mode (`-distributor.otel-metric-suffixes-enabled`).
+- Prometheus: Fix issue where negation of native histograms (e.g. `-some_native_histogram_series`) did nothing.
+- Prometheus: Always return unknown hint for first sample in non-gauge native histograms chunk to avoid incorrect counter reset hints when merging chunks from different sources.
+- Prometheus: Ensure native histograms counter reset hints are corrected when merging results from different sources.
+- PromQL: Fix issue where functions such as `rate()` over native histograms could return incorrect values if a float stale marker was present in the selected range.
+- PromQL: `round` now removes the metric name again.
+- PromQL: Fix issue where `metric might not be a counter, name does not end in _total/_sum/_count/_bucket` annotation would be emitted even if `rate` or `increase` did not have enough samples to compute a result.
+- Querier: Fix the behaviour of binary operators between native histograms and floats.
+- Querier: Fix stddev+stdvar aggregations to always ignore native histograms, and to treat Infinity consistently.
```

Comment:
```suggestion
- Querier: Fix stddev+stdvar aggregations to always ignore native histograms and to treat infinity consistently.
```

Reply:
I'm torn on this; IIUC histograms use `+Inf` to denote infinity; if it's capitalized in that context, should it be capitalized here?